core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -637,6 +637,12 @@ object KafkaMetadataLog extends Logging {
       nodeId
     )
 
+    // Print a warning if users have overridden the internal config
+    if (config.internalSegmentBytes() != null) {
+      metadataLog.error(s"Overriding ${MetadataLogConfig.INTERNAL_METADATA_LOG_SEGMENT_BYTES_CONFIG} is only supported for testing. Setting " +
```

Comment:
we should print the warnings only if the segment size is configured to small value. please consider comparing the value to `METADATA_LOG_SEGMENT_BYTES_DEFAULT`